shared/actions/settings.tsx

```diff
@@ -17,251 +16,230 @@ import {isAndroidNewerThanN, pprofDir, version} from '../constants/platform'
 import {TypedState} from '../util/container'
 
-const onUpdatePGPSettings = () =>
-  RPCTypes.accountHasServerKeysRpcPromise()
-    .then(({hasServerKeys}) => SettingsGen.createOnUpdatedPGPSettings({hasKeys: hasServerKeys}))
-    .catch(error => SettingsGen.createOnUpdatePasswordError({error}))
+const onUpdatePGPSettings = async () => {
+  try {
+    const {hasServerKeys} = await RPCTypes.accountHasServerKeysRpcPromise()
+    return SettingsGen.createOnUpdatedPGPSettings({hasKeys: hasServerKeys})
+  } catch (error) {
+    return SettingsGen.createOnUpdatePasswordError({error})
+  }
+}
 
-function* onSubmitNewEmail(state: TypedState) {
+const onSubmitNewEmail = async (state: TypedState) => {
   try {
-    yield Saga.put(SettingsGen.createWaitingForResponse({waiting: true}))
     const newEmail = state.settings.email.newEmail
-    yield RPCTypes.accountEmailChangeRpcPromise({newEmail})
-    yield Saga.put(SettingsGen.createLoadSettings())
-    yield Saga.put(RouteTreeGen.createNavigateUp())
+    await RPCTypes.accountEmailChangeRpcPromise({newEmail}, Constants.settingsWaitingKey)
+    return [SettingsGen.createLoadSettings(), RouteTreeGen.createNavigateUp()]
   } catch (error) {
-    yield Saga.put(SettingsGen.createOnUpdateEmailError({error}))
-  } finally {
-    yield Saga.put(SettingsGen.createWaitingForResponse({waiting: false}))
+    return SettingsGen.createOnUpdateEmailError({error})
   }
 }
 
-function* onSubmitNewPassword(state: TypedState, action: SettingsGen.OnSubmitNewPasswordPayload) {
+const onSubmitNewPassword = async (state: TypedState, action: SettingsGen.OnSubmitNewPasswordPayload) => {
   try {
-    yield Saga.put(SettingsGen.createWaitingForResponse({waiting: true}))
     const {newPassword, newPasswordConfirm} = state.settings.password
     if (newPassword.stringValue() !== newPasswordConfirm.stringValue()) {
-      yield Saga.put(SettingsGen.createOnUpdatePasswordError({error: new Error("Passwords don't match")}))
-      return
-    }
-    yield RPCTypes.accountPassphraseChangeRpcPromise({
-      force: true,
-      oldPassphrase: '',
-      passphrase: newPassword.stringValue(),
-    })
-    yield Saga.put(RouteTreeGen.createNavigateUp())
-    if (action.payload.thenSignOut) {
-      yield Saga.put(ConfigGen.createLogout())
+      return SettingsGen.createOnUpdatePasswordError({error: new Error("Passwords don't match")})
     }
+    await RPCTypes.accountPassphraseChangeRpcPromise(
+      {
+        force: true,
+        oldPassphrase: '',
+        passphrase: newPassword.stringValue(),
+      },
+      Constants.settingsWaitingKey
+    )
+
+    return [
+      RouteTreeGen.createNavigateUp(),
+      ...(action.payload.thenSignOut ? [ConfigGen.createLogout()] : []),
+    ]
   } catch (error) {
-    yield Saga.put(SettingsGen.createOnUpdatePasswordError({error}))
-  } finally {
-    yield Saga.put(SettingsGen.createWaitingForResponse({waiting: false}))
+    return SettingsGen.createOnUpdatePasswordError({error})
```

Comment:
Why is returning better than `Saga.put`?

Reply:
this isn't a generator anymore. Saga.put is only for sagas to dispatch